packages/lib/src/toolbar/controls/Selector/Selector.tsx

```diff
@@ -28,54 +37,136 @@ function Selector<T extends string>(props: Props<T>) {
     disabled,
     onChange,
     options,
-    optionComponent: Option,
+    optionComponent: OptionComp,
   } = props;
 
-  const { height: winHeight } = useWindowSize();
-  const menuMaxHeight = winHeight - MENU_TOP - MENU_BOTTOM;
+  const [isOpen, toggle] = useToggle();
+  const [activeIndex, setActiveIndex] = useState<number | null>(null);
+  const [selectedIndex, setSelectedIndex] = useState(() => {
+    return getAllOptions(options).indexOf(value);
```

Comment:
I have to find for the index of the selected option to properly initialise the `selectedIndex` state. This requires flatting the record when option groups are involved like in the color map drop-down.